test/dyncluster.js

```diff
@@ -0,0 +1,81 @@
+/* jslint node: true */
+/* global describe: false, before: false, after: false, it: false, afterEach: false, beforeEach: false */
+/* jshint -W030 */
+'use strict';
+
+var NATS = require ('../'),
+    nsc = require('./support/nats_server_control'),
+    should = require('should');
+
+describe('Dynamic Cluster - Connect URLs', function() {
+  // this to enable per test cleanup
+  var servers;
+  // Shutdown our servers
+  afterEach(function() {
+    nsc.stop_cluster(servers);
+    servers = [];
+  });
+
+  it('adding cluster performs update', function(done) {
+    this.timeout = 10000;
+    var route_port = 54220;
+    var port = 54221;
+
+    // start a new cluster with single server
+    servers = nsc.start_cluster([port], route_port, function() {
+      should(servers.length).be.equal(1);
+
+      // connect the client
+      var nc = NATS.connect({'port': port, 'reconnectTimeWait': 100});
+      nc.on('connect', function () {
+        // start adding servers
+        process.nextTick(function () {
+          var others = nsc.add_member_with_delay([port + 1, port + 2], route_port, 250, function () {
+            // verify that 2 servers were added
+            should(others.length).be.equal(2);
+            others.forEach(function (o) {
+              // add them so they can be reaped
+              servers.push(o);
+            });
+            // give some time for the server to send infos
+            setTimeout(function () {
+              // we should know of 3 servers - the one we connected and the 2 we added
+              should(nc.servers.length).be.equal(3);
+              done();
+            }, 1000);
+          });
+        });
+      });
+    });
+  });
+
+  it('setting protocol zero doesnt update', function(done) {
```

Comment:
Why do we want to expose and let client decide? Is that how we do it in other clients, they can suppress if they want to?